src/main/scala/mesosphere/marathon/upgrade/ReadinessBehavior.scala

```diff
@@ -85,62 +85,55 @@ trait ReadinessBehavior { this: Actor with ActorLogging =>
-          taskFn(taskId)
+      def instanceIsRunning(instanceFn: Instance => Unit): Receive = {
+        case InstanceChanged(_, `version`, `pathId`, Running, instance) => instanceFn(instance)
       }
-      taskIsRunning(
+      instanceIsRunning(
         if (runSpec.readinessChecks.isEmpty) markAsHealthyAndReady else markAsHealthyAndInitiateReadinessCheck)
     }
 
-    def taskHealthBehavior: Receive = {
+    def instanceHealthBehavior: Receive = {
       def initiateReadinessOnRun: Receive = {
-        case MesosStatusUpdateEvent(_, taskId, "TASK_RUNNING", _, `pathId`, _, _, _, `versionString`, _, _) =>
-          initiateReadinessCheck(taskId)
+        case InstanceChanged(_, `version`, `pathId`, Running, instance) => initiateReadinessCheck(instance)
       }
-      def handleTaskHealthy: Receive = {
-        case HealthStatusChanged(`pathId`, taskId, `version`, true, _, _) if !healthy(taskId) =>
-          log.info(s"Task $taskId now healthy for app ${runSpec.id.toString}")
-          healthy += taskId
-          if (runSpec.readinessChecks.isEmpty) ready += taskId
-          taskStatusChanged(taskId)
+      def handleInstanceHealthy: Receive = {
+        case InstanceHealthChanged(id, `version`, `pathId`, true) if !healthy(id) =>
+          log.info(s"Instance $id now healthy for run spec ${runSpec.id}")
+          healthy += id
+          if (runSpec.readinessChecks.isEmpty) ready += id
+          instanceStatusChanged(id)
       }
-      val handleTaskRunning = if (runSpec.readinessChecks.nonEmpty) initiateReadinessOnRun else Actor.emptyBehavior
-      handleTaskRunning orElse handleTaskHealthy
+      val handleInstanceRunning = if (runSpec.readinessChecks.nonEmpty) initiateReadinessOnRun else Actor.emptyBehavior
+      handleInstanceRunning orElse handleInstanceHealthy
     }
 
-    def initiateReadinessCheck(taskId: Instance.Id): Unit = {
-      log.debug(s"Initiate readiness check for task: $taskId")
-      val me = self
-      instanceTracker.instance(taskId).map {
-        case Some(task: Task) =>
-          for {
-            launched <- task.launched
-          } me ! ScheduleReadinessCheckFor(task, launched)
-        case _ => // TODO POD support
-      }
-    }
-
-    def readinessCheckBehavior: Receive = {
-      case ScheduleReadinessCheckFor(task, launched) =>
+    def initiateReadinessCheck(instance: Instance): Unit = {
+      def initiateReadinessCheckForTask(task: Task, launched: Task.Launched): Unit = {
         log.debug(s"Schedule readiness check for task: ${task.id}")
         ReadinessCheckExecutor.ReadinessCheckSpec.readinessCheckSpecsForTask(runSpec, task, launched).foreach { spec =>
           val subscriptionName = ReadinessCheckSubscriptionKey(task.id, spec.checkName)
           val subscription = readinessCheckExecutor.execute(spec).subscribe(self ! _)
           subscriptions += subscriptionName -> subscription
         }
+      }
+      for {
```

Comment:
Haha, I do think it's harder to read, but it's also really slow.

On Fri, Sep 9, 2016 at 10:07 AM Matthias Veit <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/upgrade/ReadinessBehavior.scala
> <https://github.com/mesosphere/marathon/pull/4343#discussion_r78214591>:
>
> >          log.debug(s"Schedule readiness check for task: ${task.id}")
> >          ReadinessCheckExecutor.ReadinessCheckSpec.readinessCheckSpecsForTask(runSpec, task, launched).foreach { spec =>
> >            val subscriptionName = ReadinessCheckSubscriptionKey(task.id, spec.checkName)
> >            val subscription = readinessCheckExecutor.execute(spec).subscribe(self ! _)
> >            subscriptions += subscriptionName -> subscription
> >          }
> > +      }
> > +      for {
>
> You hate for :)
> Changed it.
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4343/files/4cce6abfa05a4ac413248acc75987af864b30ec6#r78214591>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAweZrq_WovzWMOAhYGtzQ59b0VRaYKTks5qoZJhgaJpZM4J5N5X>
> .
>